helios-client/src/main/java/com/spotify/helios/client/tls/SshAgentContentSigner.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2015 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.helios.client.tls;
+
+import com.google.common.base.Throwables;
+
+import com.spotify.sshagentproxy.DefaultAgentProxy;
+import com.spotify.sshagentproxy.Identity;
+
+import org.bouncycastle.asn1.x509.AlgorithmIdentifier;
+import org.bouncycastle.operator.ContentSigner;
+import org.bouncycastle.operator.DefaultSignatureAlgorithmIdentifierFinder;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+
+/**
+ * Returns a signed SHA1 hash of any arbitrary data by sending it to the local SSH agent.
```

Comment:
Random note: I had a hard time figuring out that the `AgentProxy.sign()` method does the SHA1-hashing (as opposed to the content signer doing the hashing somewhere in the super class and then sending the hash for signing). It's not very apparent from the `AgentProxy` interface that it will hash and sign.